pallets/prices/src/lib.rs

```diff
@@ -0,0 +1,193 @@
+// Copyright 2021 Parallel Finance Developer.
+// This file is part of Parallel Finance.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Copyright (C) 2020-2021 Acala Foundation.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! # Prices Module
+//!
+//! ## Overview
+//!
+//! The data from Oracle cannot be used in business, prices module will do some
+//! process and feed prices for parallel. Process include:
+//!   - specify a fixed price for stable currency
+//!   - feed price in USD or related price bewteen two currencies
+//!   - lock/unlock the price data get from oracle
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::unused_unit)]
+
+use frame_support::{pallet_prelude::*, transactional};
+use frame_system::pallet_prelude::*;
+use orml_traits::{DataFeeder, DataProvider};
+use primitives::CurrencyId;
+use sp_runtime::{
+	FixedU128, FixedPointNumber
+};
+
+pub mod weights;
+
+pub use module::*;
+pub use weights::WeightInfo;
+
+pub const CURRENCY_DECIMAL: u32 = 18;
+
+pub type OraclePrice = FixedU128;
+
+pub trait PriceProvider<CurrencyId> {
```

Comment:
Good Job, I think we may need to discuss here, since we already have a trait "[PriceFeeder](https://github.com/parallel-finance/parallel/blob/master/primitives/src/lib.rs#L112)", we may need one of them

Reply:
actually in other pallets, such as pallet loans, will use "PriceFeeder" to get the price, it seems like now we need the trait "PriceProvider" to get the price, so pallet loans need to change their code too.
maybe we can simply implement one single method just like "PriceFeeder" do, and by annotating other methods, 
we could hear others views

Reply:
Is it better to add a "PriceFeeder" implementation while keeping the "PriceProvider" implementation?

Reply:
Fixed: use "PriceProvider" only.